marketplace/deployer_util/config_helper.py

```diff
@@ -531,6 +539,86 @@ def nodes(self):
     return self._nodes
 
 
+class SchemaAssistedClusterCreation:
+  """Accesses top level AssistedClusterCreation constraints."""
+
+  _ASSISTED_CC_TYPE_DISABLED = "DISABLED"
+  _ASSISTED_CC_TYPE_STRICT = "STRICT"
+  _ASSISTED_CC_TYPES = [_ASSISTED_CC_TYPE_DISABLED, _ASSISTED_CC_TYPE_STRICT]
+
+  def __init__(self, dictionary):
+    self._type = None
+    self._creation_guidance = None
+    self._gke = None
+
+    self._type = dictionary.get('type', None)
+    _must_contain(self._type, self._ASSISTED_CC_TYPES,
+                  "Invalid type of AssistedClusterCreation")
+    self._creation_guidance = dictionary.get('creationGuidance', "")
+    self._gke = _maybe_get_and_apply(dictionary, 'gke', lambda v: SchemaGke(v))
+
+    if self._type == self._ASSISTED_CC_TYPE_DISABLED and not self._creation_guidance:
+      raise InvalidSchema(
+          'assistedClusterCreation.creationGuidance must be specified when '
+          'assistedClusterCreation.type is DISABLED')
+    if self._type == self._ASSISTED_CC_TYPE_STRICT and not self._gke:
+      raise InvalidSchema('assistedClusterCreation.gke must be specified when '
+                          'assistedClusterCreation.type is STRICT')
+
+  @property
+  def type(self):
+    return self._type
+
+  @property
+  def creation_guidance(self):
+    return self._creation_guidance
+
+  @property
+  def gke(self):
+    return self._gke
+
+
+class SchemaGke:
+
+  def __init__(self, dictionary):
+    self._node_pool = None
+    node_pool = dictionary['nodePool']
+    if not isinstance(node_pool, list):
+      raise InvalidSchema('gke.nodePool must be a list')
+    self._node_pool = [SchemaNodePoolDetails(r) for r in node_pool]
+    if len(self._node_pool) != 1:
+      raise InvalidSchema('gke.nodePool supports exactly one nodePool')
+
+  @property
+  def node_pool(self):
+    return self._node_pool
+
+
+class SchemaNodePoolDetails:
+
+  def __init__(self, dictionary):
+    self._num_nodes = _must_get(dictionary, 'numNodes',
+                                'NodePoolDetails must have numNodes property')
+    self._machine_type = _must_get(
+        dictionary, 'machineType',
+        'NodePoolDetails must have machineType property')
+    if "custom-" in self._machine_type:
```

Comment:
When you update `schema.md`, you should reference both the [predefined machine types](https://cloud.google.com/compute/docs/machine-types) and the [custom machine type docs](https://cloud.google.com/compute/docs/instances/creating-instance-with-custom-machine-type#specifications) since there are too many requirements for us to document.

Reply:
Yes, will definitely do that.